pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1131,6 +1038,23 @@ async def parse_packages(self, primary_xml, filelists_xml, other_xml, file_exten
                 await packages_pb.aincrement()
                 await self.put(dc)
 
+            if settings.RPM_ITERATIVE_PARSING:
+                # if settings.RPM_USE_NEW_ITERATIVE_PARSER:
+
+                #     def _package_callback(pkg):
+                #         if not skip_srpms or pkg.arch != "src":
+                #             package_callback(pkg)
+
+                #     parser.for_each_package(_package_callback)
```

Comment:
Problem:  Our code is async and does Async Things.  This pattern (internal iterator) takes complete control of the execution and is synchronous.  Somehow we need to have the callbacks not only add things to the queue but also let the event loop run for long enough to keep the pipeline from building up.